addons/default/visiosoft/advs-module/src/Http/Controller/advsController.php

```diff
@@ -151,29 +151,41 @@ public function index($category = null, $city = null)
         $isActiveDopings = $this->adv_model->is_enabled('dopings');
 
         // Search by category slug
+        $categoryId = null;
         if ($category) {
             $categoryId = $this->category_repository->findBy('slug', $category);
-            if ($categoryId) {
-                $param['cat'] = $categoryId->id;
+            if (!$categoryId) {
+                $this->messages->error(trans('visiosoft.module.advs::message.category_not_exist'));
+                return redirect('/');
             }
-        } elseif (isset($param['cat']) && !empty($param['cat'])) {
-            $categoryId = $this->category_repository->find($param['cat']);
-            if ($categoryId) {
+            if (isset($param['cat'])) {
+                unset($param['cat']);
                 return redirect($this->fullLink(
                     $param,
                     route('adv_list_seo', [$categoryId->slug]),
                     array()
                 ));
-            } else {
+            }
+        } elseif (isset($param['cat']) && !empty($param['cat'])) {
+            $categoryId = $this->category_repository->find($param['cat']);
+            if (!$categoryId) {
                 $this->messages->error(trans('visiosoft.module.advs::message.category_not_exist'));
                 return redirect('/');
             }
+            unset($param['cat']);
```

Comment:
in both case's you are using unset, So no need to put inside the condition.

Reply:
actually i am unsetting before redirecting to remove the params from the link, and since there are two redirect functions, I had to use it twice @ozcan-durak